feat: adding marketing analytics settings

jabahamondes opened the pull request:
Note: in a following PR, I'll backfill marketing config for the existing teams.

## Problem

Currently it's very hardcoded how we connect the tables. So now there a UI in marketing settings that allows you to connect and map any table.


## Changes

- Introduced new marketing tab in web analytics with relevant routing and insights.
- Updated constants and scene configurations to include marketing features.
- Enhanced WebAnalyticsDashboard to conditionally display marketing costs based on feature flags.
- Adjusted logic to handle marketing product tab and its associated breadcrumbs.
- Updated WebPropertyFilters to hide filters when on the marketing tab


## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've added or updated the docs
- [ ] I've reached out for help from the docs team
- [x] No docs needed for this change

## How did you test this code?

Manually
<img width="1202" alt="image" src="https://github.com/user-attachments/assets/203ab086-e8a9-4ccc-990a-3648b4fa9b9a" />

<img width="1241" alt="image" src="https://github.com/user-attachments/assets/bc9c89e5-4740-435e-9235-442f3398cf69" />

We only show the sources that have the required fields mapped
<img width="1197" alt="image" src="https://github.com/user-attachments/assets/cec23355-2714-4d6b-bd43-b009b23d62fc" />
<img width="1202" alt="image" src="https://github.com/user-attachments/assets/751ba5dd-1311-405f-83e6-34c6d65c1623" />

Commit:
refactor: Update pipelineNodeNew to use source instead of kind

- Changed the parameter from 'kind' to 'source' in urls.pipelineNodeNew across multiple files.
- Updated related logic in sourceWizardLogic and DataWarehouseIntegrationChoice to accommodate the new parameter.
- Ensured consistency in handling external data sources by using lowercase values for source types.

Commit:
feat(marketing): Add Marketing Analytics configuration and settings

- Introduced  type and integrated it into the  interface.
- Added mock data for  in .
- Updated schema to include  and .
- Created new components for Marketing Analytics settings, including  and .
- Implemented logic for managing source mappings in .
- Added routes and UI elements for Marketing Analytics in the data management scene.
- Established backend support with new models and serializers for marketing analytics configuration.

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update query snapshots

Commit:
fix: Prs comments

Commit:
Merge branch 'feat/adding-marketing-settings-v2' of github.com:PostHog/posthog into feat/adding-marketing-settings-v2

Commit:
fixing snapshot

Commit:
updated snapshot

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

Commit:
refactor: Update NonNativeExternalDataSourceConfiguration to use dataWarehouseSettingsLogic

- Switched data source logic to utilize dataWarehouseSettingsLogic for fetching dataWarehouseSources.
- Maintained existing functionality by keeping sources_map from marketingAnalyticsSettingsLogic.

Commit:
Merge branch 'feat/adding-marketing-settings-v2' of github.com:PostHog/posthog into feat/adding-marketing-settings-v2

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

Commit:
fix(migrations): correct dependency for 0763_migrate_team_base_currency and add 0764_teammarketinganalyticsconfig migration

Commit:
fix(migrations): update dependency for 0764_teammarketinganalyticsconfig migration to 0763_migrate_team_base_currency

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

Commit:
feat(migrations): remove obsolete 0763_migrate_team_base_currency migration and add 0763_teammarketinganalyticsconfig model

Commit:
Purpose mistake

Commit:
fixing

jabahamondes commented on frontend/src/scenes/teamActivityDescriber.tsx:
```diff
@@ -476,6 +476,7 @@ const teamActionsMapping: Record<
     slack_incoming_webhook: () => null,
     timezone: () => null,
     revenue_analytics_config: () => null,
+    marketing_analytics_config: () => null,
```
This is also in my todo list for the following PRs

rafaeelaudibert reviewed (commented):
There's a small change I suggest you make to the schema. LGTM except that!
Will leave final approval for the @PostHog/web-analytics team

jabahamondes commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/NonNativeExternalDataSourceConfiguration.tsx:
```diff
@@ -0,0 +1,276 @@
+    columns?: { name: string; type: string }[]
+}
+
+// This is to map tables that are not natively integrated with PostHog.
+// It's a workaround to allow users to map columns to the correct fields in the Marketing Analytics product.
+// An example of native integration is the Google Ads integration.
```
I'll update it. Yes, it's very flexible and scalable to any need 👀  let's consider it!

Commit:
refactor(schema): move SourceMap type definition to schema-general and update imports

- Moved `SourceMap` type from `marketing-analytics/utils` to `schema-general.ts` for better organization.
- Updated imports in `webAnalyticsLogic.tsx` and `marketingAnalyticsSettingsLogic.ts` to reflect the new location.
- Adjusted the `_sources_map` field in the migration to ensure it defaults to a non-null value.
- Cleaned up comments in `NonNativeExternalDataSourceConfiguration.tsx` for clarity.

jabahamondes commented on frontend/src/queries/schema/schema-general.ts:
```diff
@@ -1,5 +1,6 @@
 import { DataColorToken } from 'lib/colors'
 import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { SourceMap } from 'scenes/web-analytics/tabs/marketing-analytics/utils'
```
Moved!

Commit:
fix(team): update _sources_map field to enforce non-null constraint

- Changed the _sources_map field in TeamMarketingAnalyticsConfig from null=True to null=False to ensure it always has a value.

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-marketing-settings-v2

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

lricoy reviewed (approved):
Ship it and let's see 🚀
